contextIO.go

```diff
@@ -149,5 +171,13 @@ func (c *ContextIO) DoJSON(method, q string, params url.Values, body *string) (j
 	}
 	defer response.Body.Close()
 	json, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
```

Comment:
It would be pretty easy for you to implement your own wrapper on that function to behave the way you are wanting.